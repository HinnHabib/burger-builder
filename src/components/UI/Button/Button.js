import React from "react";

import classes from "./Button.module.css";

const button = (props) => {
  const { disabled, btnType, clicked, children, ...rest } = props;
  return (
    <button
      disabled={disabled}
      className={[classes.Button, classes[btnType] || " "].join(" ")}
      onClick={clicked}
      {...rest}
    >
      {children}
    </button>
  );
};

export default button;
