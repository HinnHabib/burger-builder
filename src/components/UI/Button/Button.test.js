import React from "react";
import { render } from "@testing-library/react";
import Button from "./Button";

test("renders button with text", () => {
  const { getByText } = render(<Button disabled>Hello World</Button>);
  const button = getByText("Hello World");
  expect(button).toBeDisabled();
  expect(button).toBeInTheDocument();
});

test("render button snapshot", () => {
  jest.mock("./Button.module.css", () => ({}));
  const { container } = render(
    <Button btnType={"Success"}>Hello World</Button>
  );
  expect(container.firstChild).toMatchSnapshot();
});
