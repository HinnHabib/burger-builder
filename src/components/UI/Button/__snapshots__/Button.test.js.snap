// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button snapshot 1`] = `
<button
  class="Button Success"
>
  Hello World
</button>
`;
